wgpu-core/src/hub.rs

```diff
@@ -451,10 +451,11 @@ impl<T, I: TypedId + Copy, F: IdentityHandlerFactory<I>> Registry<T, I, F> {
         id
     }
 
-    pub fn unregister_locked(&self, id: I, guard: &mut Storage<T, I>) -> T {
-        let value = guard.remove(id).unwrap();
+    pub fn unregister_locked(&self, id: I, guard: &mut Storage<T, I>) -> Option<T> {
+        let value = guard.remove(id);
         //Note: careful about the order here!
         self.identity.free(id);
+        //Returning None is legal if it's an error ID
         value
```

Comment:
I believe we are missing a few things:
  1. the same kind of modification of `unregister()`
  2. the logic in `fn remove()` that would replace an `Error` entry with `Vacant`

Speaking of which, I think `remove` should panic if the element is `Vacant`, or the index is out of range. Currently it returns an `Option`, which is ambiguous (was it a bad id? or was it an error element? these are different cases).

Reply:
For out of range access, I purposely returned `None` in https://github.com/gfx-rs/wgpu/pull/782 to match with `VecMap` implementation.
I will try refactoring the code that makes these out of range calls.

Reply:
https://github.com/gfx-rs/wgpu/blob/f6f7210b6323e643ff0000f3a4b25b2bb911e848/wgpu-core/src/device/mod.rs#L3457

This is code that makes out of range calls to `remove()`

Reply:
uuhhh

Reply:
it seems to be quite an edge case. Maybe we can roll out a different `try_remove` method for this?